teuthology/packaging.py

```diff
@@ -369,3 +381,231 @@ def get_package_version(remote, package):
         )
 
     return installed_ver
+
+
+def _get_config_value_for_remote(ctx, remote, config, key):
+    """
+    Look through config, and attempt to determine the "best" value to use
+    for a given key. For example, given:
+
+        config = {
+            'all':
+                {'branch': 'master'},
+            'branch': 'next'
+        }
+        _get_config_value_for_remote(ctx, remote, config, 'branch')
+
+    would return 'master'.
```

Comment:
This code was ported from teuthology.task.install and had had this logic for quite some time. I can't say that I truly understand the logic behind this either, but I'm hesitant to change it.